src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchReader.cs

```diff
@@ -124,6 +136,21 @@ protected override ODataBatchOperationRequestMessage CreateOperationRequestMessa
             string atomicityGroupId = (string)this.messagePropertiesCache.GetPropertyValue(
                 ODataJsonLightBatchPayloadItemPropertiesCache.PropertyNameAtomicityGroup);
 
+            if (id != null && !this.requestIdToAtomicGroupId.ContainsKey(id))
+            {
+                this.requestIdToAtomicGroupId.Add(id, atomicityGroupId);
+            }

```

Comment:
Rather than do an explicit ContainsKey check before calling Add, why not just do:
if (id != null)
{
    this.requestIdToAtomicGroupId.TryAdd(id, atomicityGroupId);
}

Reply:
Fixed

Reply:
Should you use TryAdd(id) rather than Add(id) to avoid possible exceptions if the id already exists, or are we asserting that the id doesn't already exist? (same for below)

---
In reply to: [454179343](https://github.com/OData/odata.net/pull/1831#discussion_r454179343) [](ancestors = 454179343)

Reply:
Never mind; I see that you moved to HashSet, which is perfect.

---
In reply to: [467600534](https://github.com/OData/odata.net/pull/1831#discussion_r467600534) [](ancestors = 467600534,454179343)